server/src/main/java/org/elasticsearch/index/CompositeIndexEventListener.java

```diff
@@ -256,7 +256,7 @@ private void iterateBeforeIndexShardRecovery(
                 nextListener.beforeIndexShardRecovery(indexShard, indexSettings, future);
                 if (future.isDone()) {
                     // common case, not actually async, so just check for an exception and continue on the same thread
-                    future.get();
```

Comment:
NB subtle difference in exception-handling here, `future.get()` may throw `InterruptedException` or `ExecutionException` but `future.result()` either unwraps or re-wraps these into some kind of `RuntimeException`. Shouldn't be a problem a practice, any exception here fails the shard recovery, just calling it out for completeness.

Reply:
I think my comment on `PlainActionFuture.result` would alleviate this subtle difference.

Reply:
Not quite - now we avoid the `InterruptedException` of course but `future.get()` still applies an `ExecutionException` wrapper which `future.result()` turns into a `RuntimeException`.